server/src/main/java/org/opensearch/tasks/Task.java

```diff
@@ -172,11 +180,18 @@ private TaskInfo taskInfo(String localNodeId, boolean detailed, boolean excludeS
             status = getStatus();
         }
         if (excludeStats == false) {
+            TaskThreadUsage threadInfo = getThreadUsage();
             resourceStats = new TaskResourceStats(new HashMap<>() {
                 {
                     put(TOTAL, getTotalResourceStats());
+
+                    if (threadInfo.getThreadExecutions() > 1) {
```

Comment:
I understand that these numbers are all the same if thread executions == 1, and as a human consumer of this API it is simpler if they are omitted in that case, but if I'm writing a program to consume this output then it is probably simpler not to have this conditional behavior. @reta what do you think? Do we have precedent in the API for this type of conditional behavior?

Reply:
@andrross @reta Thanks for the feedback on this. I was also thinking that this may be confusing if there is a task that has only 1 thread at first and then forks into multiple threads and the API response will actually change for the same task. I will go ahead and make this change.